crates/language/src/target_language.rs

```diff
@@ -173,6 +185,88 @@ impl PatternLanguage {
         Self::get_language_with_parser(&mut parser, src)
     }
 
+    fn get_aliases(&self) -> &'static [&'static str] {
+        for (variant, aliases) in LANGUAGE_ALIASES {
+            if variant == self {
+                return aliases;
+            }
+        }
+        &[]
+    }
+
+    fn map_value(s: String) -> Self {
+        if let Some(lang) = Self::from_string(&s, None) {
+            return lang;
+        }
+        if let Some(lang) = Self::from_extension(&s) {
+            return lang;
+        }
+        let s = s.to_lowercase();
+        for (variant, aliases) in LANGUAGE_ALIASES {
+            for alias in *aliases {
+                if *alias == s {
+                    return *variant;
+                }
+            }
+        }
+        Self::Universal
+    }
+
+    fn possible_values() -> Vec<&'static str> {
+        let mut unique_variants = std::collections::HashSet::new();
+        let mut ordered_variants = Vec::new();
+        for lang in PatternLanguage::enumerate() {
+            let lang_name = lang.to_str();
+            if unique_variants.insert(lang_name) {
+                ordered_variants.push(lang_name);
+            }
+            let aliases = lang.get_aliases();
+            for alias in aliases {
+                if unique_variants.insert(alias) {
+                    ordered_variants.push(alias);
+                }
+            }
+            for ext in lang.get_file_extensions() {
+                if unique_variants.insert(ext) {
+                    ordered_variants.push(ext);
+                }
+            }
+        }
+        ordered_variants
+    }
+
+    pub fn value_parser() -> MapValueParser<PossibleValuesParser, fn(String) -> Self> {
+        PossibleValuesParser::new(Self::possible_values()).map(|s| Self::map_value(s))
+    }
+
+    pub fn to_str(&self) -> &'static str {
```

Comment:
This seems redundant with https://github.com/getgrit/gritql/blob/5e7ca26133fd2d847f73966ca5c45b121b132faf/crates/language/src/target_language.rs#L86. Why do we need to list again?

Reply:
I guess we don't necessarily - I added it because it was mapping more "literally" - i.e. closely matching the lowercase clap/serde transformation (i.e. `JavaScript => "javascript"`) and any name/renames like (i.e. `Tsx => "js"`, and `MarkdownBlock => "markdown"`) but it wasn't honouring the skipped fields, which you also mentioned. I thought "javascript" was a good alias for "js" so this would be a "default" way to include that alias without explicitly adding it to `LANGUAGE_ALIASES` but perhaps explict is better anyway and I can remove this `to_str` function at the same time?

Reply:
I'm fine with the existing format names (`js`) being used as the names we list in the output so it looks like this can be dropped.

Reply:
removed `to_str()` and replaced its usage with `to_string()` (using `impl fmt::Display for TargetLanguage`)
in: adc787b